client/landing/gutenboarding/onboarding-block/edit.tsx

```diff
@@ -37,40 +44,81 @@ const OnboardingEdit: FunctionComponent< BlockEditProps< Attributes > > = () =>
 		window.scrollTo( 0, 0 );
 	}, [ pathname ] );
 
+	const canUseDesignSelection = useCallback( (): boolean => {
+		return ! ( ! siteVertical && ! siteTitle && ! wasVerticalSkipped );
+	}, [ siteTitle, siteVertical, wasVerticalSkipped ] );
+
+	const canUseStyleStep = useCallback( (): boolean => {
+		return !! selectedDesign && isEnabled( 'gutenboarding/style-preview' );
+	}, [ selectedDesign ] );
+
+	const canUseCreateSiteStep = useCallback( (): boolean => {
+		return isCreatingSite || isRedirecting;
+	}, [ isCreatingSite, isRedirecting ] );
+
+	const getLatestStepPath = (): string => {
+		if ( ! canUseDesignSelection() ) {
+			return makePath( Step.IntentGathering );
+		}
+		if ( ! canUseStyleStep ) {
+			return makePath( Step.DesignSelection );
```

Comment:
Ahh you're missing the parentheses to invoke the canUseStyleStep callback here!

VALID JAVASCRIPT

Reply:
D'oh! I've fixed this up and re-ordered it a little to remove the negations so that it's a bit more readable.